corelib-storage/src/main/java/eu/europeana/corelib/solr/entity/WebResourceImpl.java

```diff
@@ -356,10 +356,10 @@ public Integer getEbucoreAudioChannelNumber() {
     @Override
     public String getEbucoreDuration() {
         if (webResourceMetaInfo != null) {
-            if (webResourceMetaInfo.getAudioMetaInfo() != null) {
+            if (webResourceMetaInfo.getAudioMetaInfo() != null && webResourceMetaInfo.getAudioMetaInfo().getDuration() != null) {
                 return Long.toString(webResourceMetaInfo.getAudioMetaInfo().getDuration());
             }
-            if (webResourceMetaInfo.getVideoMetaInfo() != null) {
+            if (webResourceMetaInfo.getVideoMetaInfo() != null && webResourceMetaInfo.getAudioMetaInfo().getDuration() != null) {
```

Comment:
I suppose line 362 should be:
... webResourceMetaInfo.get**Video**MetaInfo().getDuration() != null) {

I'll commit it